src/portable/portablectl.c

```diff
@@ -1049,9 +1049,29 @@ static int is_image_attached(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
-        r = bus_call_method(bus, bus_portable_mgr, "GetImageState", &error, &reply, "s", image);
+        method = strv_isempty(arg_extension_images) ? "GetImageState" : "GetImageStateWithExtensions";
+
+        r = bus_message_new_method_call(bus, &m, bus_portable_mgr, method);
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = sd_bus_message_append(m, "s", image);
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = attach_extensions_to_message(m, arg_extension_images);
```

Comment:
Is this OK to be called for `GetImageState`?

Reply:
yes, this helper is made to make that simpler, it will do nothing if the strv is empty

Reply:
Thanks.

Reply:
Hm, it calls `sd_bus_message_open_container(m, 'a', "s"); sd_bus_message_close_container(m);` even if `arg_extension_images` is empty. I cannot remember right now and not sure, is it really does nothing to the message?

I think it is simpler to move this to the if block below.

Reply:
It returns immediately as the first thing if the strv is empty:

https://github.com/systemd/systemd/blob/f338a496e48392929400916ebb0eb9819dec1e2b/src/portable/portablectl.c#L98

Reply:
Ouch. Indeed.